content/kubermatic/main/architecture/concept/kkp-concepts/kkp-security/iam/_index.en.md

```diff
@@ -0,0 +1,54 @@
++++
+title = "Kubermatic Kubernetes Platform IAM"
+date = 2023-02-07T12:15:02+02:00
+weight = 20
+
++++
+
+KKP component (dashboard, user clusters ...) use OpenID Connect (OIDC) protocol for authentication and RBAC for authorization.
+By default, KKP provides [Dex](#dex) as OIDC provider, but you can configure your own provider. For more information, 
+please refer to the [OIDC provider]({{< ref "../../../../../tutorials-howtos/OIDC-Provider-Configuration" >}}) chapter.
+
+## Authentication with Dex
+[Dex](https://dexidp.io/) is an identity service that uses OIDC to drive authentication for KKP components. It acts as a
+portal to other identity providers through [connectors](https://dexidp.io/docs/connectors/). This lets Dex defer
+authentication to [LDAP](https://dexidp.io/docs/connectors/ldap/) servers, [SAML](https://dexidp.io/docs/connectors/saml/)
+providers, or established identity providers like [GitHub](https://dexidp.io/docs/connectors/github/), [Google](https://dexidp.io/docs/connectors/google/),
+and [Microsoft](https://dexidp.io/docs/connectors/microsoft/).  Check out the [Dex documentation](https://dexidp.io/docs/connectors/)
+for a list of available providers.
+
+To configure Dex connectors edit `.dex.connectors` in the `values.yaml`
+
+Example to set up Github connector:
+```
+dex:
+    ingress:
+    [...]
+    connectors:
+    - type: github
+        id: github
+        name: GitHub
+        config:
+        clientID: <client_id>
+        clientSecret: <client_secret>
+        redirectURI: https://<your-kkp.domain>/dex/callback
+        orgs:
+        - name: <your-github-org>
+```
+
+And apply the changes to the cluster:
+**Helm 3**
+
+```bash
+helm --namespace oauth upgrade --install --wait --values /path/to/your/helm-values.yaml oauth charts/oauth/
+```
```

Comment:
This is already explained in the installation instructions (https://docs.kubermatic.com/kubermatic/main/installation/install-kkp-ce/#prepare-configuration), albeit in less detail. Should we really repeat it here? Does it offer new information that the user didn't have before?

Reply:
_If_ we want to keep this section, I think we need to go into more detail. Like having dedicated sections for the individual connectors like GitHub or Google and explain what values you need, how to configure them, etc. The status quo IMHO is that we're not giving enough information to be really helpful on how to set this up.

Reply:
The idea was to make the most popular identity provider searchable. It feels a bit cumbersome to me to add them in the table of https://docs.kubermatic.com/kubermatic/main/installation/install-kkp-ce/#prepare-configuration.

Moreover in https://docs.kubermatic.com/kubermatic/main/architecture/role-based-access-control/groups-support/#considerations we mention that:

> Some connectors like the [Google connector](https://dexidp.io/docs/connectors/google/) require [special configuration](https://dexidp.io/docs/connectors/google/#fetching-groups-from-google) to be able to read group information

So I would like a provide a quick way to update the dex settings, but maybe it's more a development hack, and recommended method should go through the installer?


Concerning the fact that we don't give enough information, I am not sure we can do better without copying the official dex doc, and I would prefer to link to the official documentation. If we keep this section, I can add something like `To know how to set up fields, please refer to the dex documentation`